src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -259,7 +247,33 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
                 return Task.FromResult(1);
             }
 
-            return EntryPointExecutor.Execute(assembly, args, host.ServiceProvider);
+            return EntryPointExecutor.Execute(assembly, args, host.ServiceProvider)
+                        .ContinueWith(t =>
+                        {
+                            t.Exception?.Handle(e =>
```

Comment:
We lock using GetAwaiter().GetResult() which doesn't wrap exception into AggregateException